modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -229,18 +227,20 @@ package object engine {
     getS(id).flatMap(
       _.map { seq =>
         seq match {
-          case Sequence.State.Final(_,_) => unit
-          case _                         => {
+          case Sequence.State.Final(_, _) =>
+            // The sequence is marked as completed here
+            putS(id)(seq) *> send(finished(id))
```

Comment:
I'm a bit unsure that this is the correct solution. But otherwise the engine gets into an infinite loop when calling run on an empty sequence